py3status/modules/stopwatch.py

```diff
@@ -0,0 +1,154 @@
+
+Format placeholders:
+    {stopwatch} display hours:minutes:seconds
+
+@author Jonas Heinrich
+
+SAMPLE OUTPUT
+{'full_text': 'Stopwatch 0:01:00'}
+
+running
+[
+    {'full_text': 'Stopwatch '},
+    {'color': '#00FF00', 'full_text': '0'},
+    {'full_text': ':'},
+    {'color': '#00FF00', 'full_text': '00'},
+    {'full_text': ':'},
+    {'color': '#00FF00', 'full_text': '54'},
+]
+
+paused
+[
+    {'full_text': 'Stopwatch '},
+    {'color': '#FFFF00', 'full_text': '0'},
+    {'full_text': ':'},
+    {'color': '#FFFF00', 'full_text': '00'},
+    {'full_text': ':'},
+    {'color': '#FFFF00', 'full_text': '54'},
+]
+"""
+
+from time import time
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    format = 'Stopwatch {stopwatch}'
+
+    def post_config_hook(self):
+        self.running = False
+        self.time_start = None
+        self.time_state = None
+        self.color = None
+        self.paused = False
+
+    def stopwatch(self):
+
+        def make_2_didget(value):
+            value = str(value)
+            if len(value) == 1:
+                value = '0' + value
+            return value
+
+        if self.running:
+            t = int(time() - self.time_start)
+        else:
+            if self.time_state:
+                t = self.time_state
+            else:
+                t = 0
+
+        # Hours
+        hours, t = divmod(t, 3600)
+        # Minutes
+        mins, t = divmod(t, 60)
+        # Seconds
+        seconds = t
+
+        if self.running:
+            cached_until = self.py3.time_in(0, offset=1)
+        else:
+            cached_until = self.py3.CACHE_FOREVER
+
+        composites = [
+            {
+                'full_text': str(hours),
+                'color': self.color,
+                'index': 'hours',
+            },
+            {
+                'color': '#CCCCCC',
+                'full_text': ':',
+            },
+            {
+                'full_text': make_2_didget(mins),
+                'color': self.color,
+                'index': 'mins',
+            },
+            {
+                'color': '#CCCCCC',
+                'full_text': ':',
+            },
+            {
+                'full_text': make_2_didget(seconds),
+                'color': self.color,
+                'index': 'seconds',
+            },
+        ]
+
+        stopwatch = self.py3.composite_create(composites)
+
+        return {
+            'cached_until': cached_until,
+            'full_text':
+            self.py3.safe_format(self.format, {'stopwatch': stopwatch})
+        }
+
+    def on_click(self, event):
+        button = event['button']
+
+        if button == 1:
+            if self.running:
+                # pause stopwatch
+                self.running = False
+                self.paused = True
+                self.time_state = int(time() - self.time_start)
+                self.color = '#FFFF00'
```

Comment:
Remove this hardcoded colors by using this instead:
* `self.color = self.py3.COLOR_GOOD`
* `self.color = self.py3.COLOR_BAD`

This would allows users to change colors via...
```
# change good/bad for all modules config
general {
    color_good = '#ccffcc'
    color_bad = '#ffccff'
}
```
OR
```
# change good/bad for this per-module config
stopwatch {
    color_good = '#ccffcc'
    color_bad = '#ffccff'
}
```

Reply:
fixed :)